Fixed #4474 - function as defaults for SQLite columns requires brackets

pax0r opened the pull request:
This pull request fixes #4474

zzzeek commented on test/dialect/test_sqlite.py:
```diff
@@ -462,6 +463,21 @@ def test_boolean_default(self):
             [(False,), (True,)],
         )
 
+    @testing.provide_metadata
```
OK, I see this imitates the style that was there, which I'd like to change, but since you faithfully imitated what was there I gues that's on me...

pax0r replied:
Sorry, as you guest I just imitated test code from above as I haven't delve to much into the codebase.

zzzeek replied:
not your fault!

sqla-tester reviewed (commented):
OK, this is **sqla-tester** setting up my work to try to get revision 05a8f82bc0d1f27370f6db0c47dc2c7d4a13b5e6 of this pull request into gerrit so we can run tests and reviews and stuff

pax0r committed:
Fixed #4474 - function as defaults for SQLite columns requires brackets

This pull request fixes #4474

Closes: #4475
Pull-request: https://github.com/sqlalchemy/sqlalchemy/pull/4475
Pull-request-sha: 05a8f82bc0d1f27370f6db0c47dc2c7d4a13b5e6

Change-Id: I888c69bfba5a5ab8ffa420f512557ac311a36b31

sqla-tester commented:
New Gerrit review created for change 05a8f82bc0d1f27370f6db0c47dc2c7d4a13b5e6: https://gerrit.sqlalchemy.org/#/c/sqlalchemy/sqlalchemy/+/1121

sqla-tester reviewed (commented):
OK, this is **sqla-tester** setting up my work to try to get revision 0af3238c69c2610333cf62082c12047d45c31ce0 of this pull request into gerrit so we can run tests and reviews and stuff

sqla-tester commented:
Patchset 0af3238c69c2610333cf62082c12047d45c31ce0 added to existing Gerrit review https://gerrit.sqlalchemy.org/#/c/sqlalchemy/sqlalchemy/+/1121

zzzeek commented:
yup keep pushing to the PR here, I've integrated evrything so folks don't have to worry about gerrit anymore :)

pax0r commented:
I honestly have no idea why jenkins failed to this one...

zzzeek commented:
> I honestly have no idea why jenkins failed to this one...

again no worries, that's a stability issue on the jenkins side as one of the mysql containers has been misbehaving in recent builds, this build essentially passed

sqla-tester commented on lib/sqlalchemy/dialects/sqlite/base.py:
```diff
@@ -468,15 +468,16 @@ def set_sqlite_pragma(dbapi_connection, connection_record):
-from ...engine import reflection
-from ...sql import compiler
+from sqlalchemy import exc
+from sqlalchemy import processors
```
**mike bayer** (zzzeek) wrote:

this all has to be relative imports

sqla-tester commented on test/dialect/test_sqlite.py:
```diff
@@ -462,6 +463,21 @@ def test_boolean_default(self):
+    def test_function_default(self):
+        t = Table(
+            "t",
+            self.metadata,
```
**mike bayer** (zzzeek) wrote:

will update these

sqla-tester reviewed (commented):
**mike bayer** (zzzeek) wrote:

(3 comments)

* /COMMIT_MSG (line 7): this was submitted against 1.2 please use master in the future, we backport to 1.2

sqla-tester commented on lib/sqlalchemy/dialects/sqlite/base.py:
```diff
@@ -908,6 +909,8 @@ def get_column_specification(self, column, **kwargs):
         colspec = self.preparer.format_column(column) + " " + coltype
         default = self.get_column_default_string(column)
         if default is not None:
+            if isinstance(column.server_default.arg, functions.Function):
```
**mike bayer** (zzzeek) wrote:

shouldn't we be rendering parenthesis around *any* expression?  am adding tests now

sqla-tester reviewed (commented):
**mike bayer** (zzzeek) wrote:

(1 comment)

sqla-tester commented:
**mike bayer** (zzzeek) wrote:

Code-Review+2 Workflow+1

sqla-tester commented:
Gerrit review https://gerrit.sqlalchemy.org/1123 has been **merged**. Congratulations! :)

sqla-tester commented:
**mike bayer** (zzzeek) wrote:

Code-Review+2 Workflow+1

sqla-tester commented:
Gerrit review https://gerrit.sqlalchemy.org/1121 has been **merged**. Congratulations! :)
